src/instrumenter/annotations.ts

```diff
@@ -137,161 +149,210 @@ export class SyntaxError extends Error {
+                const [errStartOff, errLength] = offsetBy(
+                    [e.location.start.offset, e.location.end.offset],
+                    exprLoc
                 );
-            }
 
-            try {
-                const exprNode = parseExpr(exprOrig);
-
-                result.push(
-                    new Annotation(
-                        raw,
-                        target,
-                        annotationOrig,
-                        exprOrig,
-                        exprNode,
-                        annotationType,
-                        annotationLabel,
-                        exprLoc,
-                        annotationLoc,
-                        commentLoc
-                    )
-                );
+                const errRange = rangeToLocRange(errStartOff, errLength, source);
+                const original = exprOrig.slice(errStartOff - 10, errStartOff + errLength + 20);
 
-                rx.lastIndex = match.index + annotationOrig.length;
-            } catch (e) {
-                if (e instanceof ExprPEGSSyntaxError) {
-                    // Compute the syntax error offset relative to the start of the file
-                    const [errStartOff, errLength] = offsetBy(
-                        [e.location.start.offset, e.location.end.offset],
-                        exprLoc
-                    );
+                throw new SyntaxError(e.message, original, errRange);
+            }
 
-                    const errRange = rangeToLocRange(errStartOff, errLength, source);
-                    const original = exprOrig.slice(errStartOff - 10, errStartOff + errLength + 20);
+            throw e;
+        }
+    }
 
-                    throw new SyntaxError(e.message, original, errRange);
-                }
+    private validateAnnotation(target: AnnotationTarget, annotation: Annotation, source: string) {
+        if (target instanceof ContractDefinition) {
+            if (annotation.type !== AnnotationType.Invariant) {
+                throw new UnsupportedByTargetError(
+                    `Annotation type "${annotation.type}" is not applicable to a target`,
+                    annotation.original,
+                    annotation.annotationFileLoc(source)
+                );
+            }
 
-                throw e;
+            if (target.kind === ContractKind.Interface || target.kind === ContractKind.Library) {
+                throw new UnsupportedByTargetError(
+                    `Unsupported contract annotations on ${target.kind} ${target.name}`,
+                    annotation.original,
+                    annotation.annotationFileLoc(source)
+                );
+            }
+        } else if (target instanceof FunctionDefinition) {
+            if (annotation.type !== AnnotationType.IfSucceeds) {
+                throw new UnsupportedByTargetError(
+                    `Annotation type "${annotation.type}" is not applicable to a target`,
```

Comment:
Small rewording:

"${annotation.type}" annotation is not applicable to functions

Reply:
Addressed in https://github.com/ConsenSys/scribble/pull/5/commits/3b1a65c0b0e464efe1aaada050bfe36bd5592678.